src/components/RouteChangeHandler/RouteChangeHandler.js

```diff
@@ -81,7 +82,17 @@ class RouteChangeHandler extends Component {
 
   componentDidUpdate(prevProps) {
     // when changing from one page to another
-    if (this.props.location !== prevProps.location) {
+    if (this.props.location.pathname !== prevProps.location.pathname) {
```

Comment:
This was actually a hidden bug. The location object is new (with strict equals) even when the pathname doesn't change.